pkg/maps/lbmap/ipv4.go

```diff
@@ -437,6 +475,92 @@ func (v *Backend4Value) ToHost() BackendValue {
 	return &h
 }
 
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapValue
+type Backend4ValueV2 struct {
+	Address   types.IPv4      `align:"address"`
+	Port      uint16          `align:"port"`
+	Proto     u8proto.U8proto `align:"proto"`
+	Flags     uint8           `align:"flags"`
+	ClusterID uint8           `align:"cluster_id"`
+	Pad       pad3uint8       `align:"pad"`
+}
+
+func NewBackend4ValueV2(addrCluster cmtypes.AddrCluster, port uint16, proto u8proto.U8proto, state loadbalancer.BackendState) (*Backend4ValueV2, error) {
+	addr := addrCluster.Addr()
+	if !addr.Is4() {
+		return nil, fmt.Errorf("Not an IPv4 address")
+	}
+
+	clusterID := addrCluster.ClusterID()
+	if addrCluster.ClusterID() > cmtypes.ClusterIDMax {
+		return nil, fmt.Errorf("ClusterID %d is too large. ClusterID > %d is not supported with Backend4ValueV2", clusterID, cmtypes.ClusterIDMax)
+	}
+
+	flags := loadbalancer.NewBackendFlags(state)
+
+	val := Backend4ValueV2{
+		Port:      port,
+		Proto:     proto,
+		Flags:     flags,
+		ClusterID: uint8(clusterID),
+	}
+
+	ip4Array := addr.As4()
+	copy(val.Address[:], ip4Array[:])
+
+	return &val, nil
+}
+
+func (v *Backend4ValueV2) String() string {
+	vHost := v.ToHost().(*Backend4ValueV2)
+	return fmt.Sprintf("%s://%s", vHost.Proto, cmtypes.AddrClusterFrom(vHost.Address.Addr(), uint32(vHost.ClusterID)).String())
+}
+
+func (v *Backend4ValueV2) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(v) }
+
+func (b *Backend4ValueV2) GetAddress() net.IP { return b.Address.IP() }
+func (b *Backend4ValueV2) GetIPCluster() cmtypes.AddrCluster {
+	return cmtypes.AddrClusterFrom(b.Address.Addr(), uint32(b.ClusterID))
+}
+func (b *Backend4ValueV2) GetPort() uint16 { return b.Port }
+func (b *Backend4ValueV2) GetFlags() uint8 { return b.Flags }
+
+func (v *Backend4ValueV2) ToNetwork() BackendValue {
+	n := *v
+	n.Port = byteorder.HostToNetwork16(n.Port)
+	return &n
+}
+
+// ToHost converts Backend4Value to host byte order.
+func (v *Backend4ValueV2) ToHost() BackendValue {
+	h := *v
+	h.Port = byteorder.NetworkToHost16(h.Port)
+	return &h
+}
+
+type Backend4V3 struct {
+	Key   *Backend4KeyV2
+	Value *Backend4ValueV2
+}
```

Comment:
It would be nice if the version indicators in key and value would be in sync with the map version, so maybe rename as `Backend4KeyV2` and `Backend4ValueV3` (and the same for IPv6 versions)?

Reply:
Ok, I'll fix.

Reply:
Fixed